### ISSUE TITLE
Enable plugin system

### ISSUE THREAD

**Issue by kaichaosun** [holochain/launcher-tauri#42]
**Is your feature request related to a problem? Please describe.**
Ad4m starts a GraphQL service and a consistent API to wrap the web3 backend like holochain. As an ad4m developer, I want holochain users to use ad4m service directly in launcher. 

**Describe the solution you'd like**
External binaries can be downloaded with provided link, and run the binary when restart the launcher

**Describe alternatives you've considered**

Now the standalone ad4m service starts its own holochain, lair-keystore, and holochain cli processes.

**Additional context**

From @guillemcordoba 

> There was a discussion about this some time ago, the discussion revolved around allowing custom binaries to accompany a happ in the happ store, that the launcher would download and run
But I don't think that conversation was resolved.... There were some security considerations to take into account that made it a complex topic
I will ask Art if there are new developments on this front

and 

Guillem also contacted Art and unfortunately things haven't moved much, and we are not yet at a point where the concerns can be resolved. Here's what @artbrock  said:
> I have some concers about this breaking sandboxing membranes. I think we need to have a discussion for making permissions explicit, similar to how you can enable/disable cell phone permissions for apps (like access contacts, location, manage calls, etc.).  We probably have things like (execute external file(s), read from / write to file system (in assigned directories), mapping which apps/cells get which of these permission, etc.)

You may also find this issue interesting, @jdeepee @lucksus

**Comment by alexisoceann** [comment]
Hi @kaichaosun, it's nice to see more interest in this feature! I proposed a similar solution of creating an API for the Launcher (#40) so that it can be installed and used by the frontend, while still allowing for bridging. This way, the Launcher can install and operate in the background, without the user having to interact with it. It's facing the same roadblocks, but hopefully it'll progress soon!

**Comment by mattyg** [comment]
I'm also very interested in making this work. 

I was thinking another option could be to use containers as the "plugins". Podman would handle constraining the access of the container to the underlying system, and you could use container image hashes to allow or report specified containers. Then Tauri can expose the container to your frontend so you can call its internal api.

**Comment by guillemcordoba** [comment]
@mattyg any use case you have in mind? I'm very wary of the complexity that this introduces in the launcher, mostly when certain apps are tied to certain plugins.

**Comment by mattyg** [comment]
> @mattyg any use case you have in mind? I'm very wary of the complexity that this introduces in the launcher, mostly when certain apps are tied to certain plugins.

Yeah I hear that. Curious to hear your thoughts on #125 which I think is a simpler / cleaner way to support my immediate use case.

I'd like to generate screenshots of external websites. In a client-server app you would run a server that has a headless browser to request the external website, render it and generate an image. I played around with using [this serivce](https://persado.github.io/dreamcatcher/) which works great. But then I couldn't release it via the launcher.

The other use case is resource-heavy services like elasticsearch or video transcoding. Though I'd expect these to be run by power users / service providers, so maybe not important to enable through the general-purpose launcher.
